lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2704,10 +2742,150 @@ SILFunction *PartitionCloner::lookupSendReceiveFunction(StringRef fnName,
+  // %zeroScalarTensor = builtin "__tfop_Const,...
+  // Both operands have type $TensorHandle<Builtin.Int64>
+  // %eq = builtin "__tfop_Equal,$in,$in(%caseId, %zeroScalarTensor)
+  // %cond = builtin "tf_tensor_to_i1"(%eq : $TensorHandle<Builtin.Int1>)
+  // cond_br %cond, bb1, bb2
+  auto BA = getBuilder(); // Builder for accelerator.
+  auto loc = remapLocation(getUserSourceLocation(inst->getDebugLocation()));
+  auto acceleratorCaseId =
+      createAcceleratorReceive(BA, loc, remapType(intFieldSILType),
+                               /*isScalar*/ true, nextSendID, FP.configuration);
+
+  assert(inst->getNumCases() == 2 &&
+         "switch enums with 1 or >2 cases will be supported later.");
+  // Create a scalar const tensor 0, to compare with case id.
+  BuiltinInst *zeroConstTensor = nullptr;
+  {
+    auto *caseZero = BA.createIntegerLiteral(loc, intFieldSILType, 0);
+    SmallVector<SILValue, 4> operands;
+    std::string constOpName = "__tfop_Const";
+    createConstTensorAttrsOnAccel(caseZero, loc, ctx, BA, constOpName,
+                                  operands);
+    FP.configuration.handleDevicePlacement(
+        "Const", /*opDevice*/ getDeviceString(DeviceType::ALL), BA, loc,
+        operands, constOpName);
+    auto subMap = getSingleSubstitutionMapForElementType(
+        intFieldSILType.getASTType(), ctx);
+    auto tensorHandleInt64Ty =
+        convertElementTypeToTensorValueType(intFieldSILType);
+    zeroConstTensor = BA.createBuiltin(loc, ctx.getIdentifier(constOpName),
+                                       tensorHandleInt64Ty, subMap, operands);
+  }
+
+  // Cast to a TensorHandle<Builtin.Int1> value
+  BuiltinInst *condBrOperand = nullptr;
+  {
+    // Omit the metatype attr T for simplicity, and TF graphDef compiler can
+    // infer the type.
+    std::string equalOpName = "__tfop_Equal,$in,$in";
+    auto boolFieldSILType = getStdlibNumericTypeFromDecl(ctx.getBoolDecl());
+    auto tensorHandleI1Ty =
+        convertElementTypeToTensorValueType(boolFieldSILType);
+    SmallVector<SILValue, 4> operands = {acceleratorCaseId, zeroConstTensor};
+    FP.configuration.handleDevicePlacement(
+        "Equal", /*opDevice*/ getDeviceString(DeviceType::ALL), BA, loc,
+        operands, equalOpName);
+    auto subMap = getSingleSubstitutionMapForElementType(
+        boolFieldSILType.getASTType(), ctx);
+    auto *equalComparisonInst =
+        BA.createBuiltin(loc, ctx.getIdentifier(equalOpName), tensorHandleI1Ty,
+                         subMap, operands);
+    condBrOperand =
+        BA.createBuiltin(loc, ctx.getIdentifier("tf_tensor_to_i1"),
+                         boolFieldSILType, {}, {equalComparisonInst});
+  }
+
+  // The args for the true and false BBs are always sent from host to
+  // accelerator, so no BB args for them here.
+  BA.createCondBranch(
+      loc, condBrOperand, getOpBasicBlock(inst->getCase(0).second),
+      /*TrueArgs*/ OperandValueArrayRef(),
```

Comment:
Why not just use `{}` here?

Reply:
createCondBranch() has two overloaded methods that can take {}, so using {} would cause ambiguity.